avm/src/vm/opcode.rs

```diff
@@ -3076,18 +3117,86 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
   }
   io!(httplsn => fn(_args, hand_mem) {
     Box::pin(async move {
-      let port_num = Program::global().http_port;
-      let addr = SocketAddr::from(([0, 0, 0, 0], port_num));
-      let make_svc = make_service_fn(|_conn| async { Ok::<_, Infallible>(service_fn(http_listener)) });
-
-      let bind = Server::try_bind(&addr);
-      match bind {
-        Ok(server) => {
-          let server = server.serve(make_svc);
+      match &Program::global().http_config {
+        HttpType::HTTP(http) => {
+          let port_num = http.port;
+          let addr = SocketAddr::from(([0, 0, 0, 0], port_num));
+          let make_svc = make_service_fn(|_conn| async { Ok::<_, Infallible>(service_fn(http_listener)) });
+
+          let bind = Server::try_bind(&addr);
+          match bind {
+            Ok(server) => {
+              let server = server.serve(make_svc);
+              tokio::spawn(async move { server.await });
+              println!("HTTP server listening on port {}", port_num);
```

Comment:
should we use `eprintln` instead?

Reply:
I have been unsure, myself. It's not exactly an error, which is why I sent it to `stdout` instead of `stderr`. I wish there was an `stddbg` for this kind of thing. ;)